src/config/files.c

```diff
@@ -149,48 +151,53 @@ gchar*
 files_get_config_path(const char* const config_base)
 {
     gchar* xdg_config = _files_get_xdg_config_home();
-    GString* file_str = g_string_new(xdg_config);
-    g_string_append(file_str, "/profanity/");
-    g_string_append(file_str, config_base);
-    gchar* result = g_strdup(file_str->str);
+    gchar* result = g_strdup_printf("%s/profanity/%s", xdg_config, config_base);
     g_free(xdg_config);
-    g_string_free(file_str, TRUE);
-
     return result;
 }
 
 gchar*
 files_get_data_path(const char* const data_base)
 {
     gchar* xdg_data = _files_get_xdg_data_home();
-    GString* file_str = g_string_new(xdg_data);
-    g_string_append(file_str, "/profanity/");
-    g_string_append(file_str, data_base);
-    gchar* result = g_strdup(file_str->str);
+    gchar* result = g_strdup_printf("%s/profanity/%s", xdg_data, data_base);
     g_free(xdg_data);
-    g_string_free(file_str, TRUE);
-
     return result;
 }
 
 gchar*
 files_get_account_data_path(const char* const specific_dir, const char* const jid)
 {
     gchar* data_dir = files_get_data_path(specific_dir);
-    GString* result_dir = g_string_new(data_dir);
-    g_free(data_dir);
-
     gchar* account_dir = str_replace(jid, "@", "_at_");
-    g_string_append(result_dir, "/");
-    g_string_append(result_dir, account_dir);
-    g_free(account_dir);
 
-    gchar* result = g_strdup(result_dir->str);
-    g_string_free(result_dir, TRUE);
+    gchar* result = g_strdup_printf("%s/%s", data_dir, account_dir);
+
+    g_free(account_dir);
+    g_free(data_dir);
 
     return result;
 }
 
+gchar*
+files_file_in_account_data_path(const char* const specific_dir, const char* const jid, const char* const file_name)
```

Comment:
Hmm is there `get` missing? `files_*get*_file_in_account_data_path` ?

Reply:
Well, I explicitly didn't add the `get` since it differs in its behavior to all the other functions as it creates the necessary path as well ... I would've liked to name it `file_in_account_data_path` but then the prefix wouldn't fit anymore ...

The name is bad, that's true ... but `files_get_file_in_account_data_path_and_create_path` is even worse :D ... and `files_get_file_in_account_data_path` would make me expect to have the same behavior than the other getters, to only construct the string ... 

We could also change the behavior of the other getters to create the path as well!?

If you want me to add

Reply:
I see!
I'm undecided what's best :D 
Maybe just leave it as you have it right now :) I will make a note to check the naming of all the functions in that file on my todolist.